dropwizard-logging/src/main/java/io/dropwizard/logging/BootstrapLogging.java

```diff
@@ -0,0 +1,67 @@
+package io.dropwizard.logging;
+
+import ch.qos.logback.classic.Level;
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.filter.ThresholdFilter;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.core.ConsoleAppender;
+import ch.qos.logback.core.encoder.LayoutWrappingEncoder;
+
+import javax.annotation.concurrent.GuardedBy;
+import java.util.TimeZone;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+/**
+ * Utililty class to configure logging before the dropwizard yml
+ * configuration has been read, parsed, and the provided logging
+ * strategy has been applied.
+ * <p/>
+ * N.B. The methods in this class have run once semantics,
+ * multiple calls are impodent
+ */
+public class BootstrapLogging {
+
+    @GuardedBy("boostrappingLock")
+    private static boolean bootstrapped = false;
+    private static Lock bootstrappingLock = new ReentrantLock();
```

Comment:
Could we me make the lock final? I think it's better to always lock on final fields as a safety measure.
Moreover, this field is not mutated.